engine/schema/src/main/java/com/cloud/storage/dao/VMTemplateDaoImpl.java

```diff
@@ -422,6 +422,10 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
         InactiveUnremovedTmpltSearch.and("removed", InactiveUnremovedTmpltSearch.entity().getRemoved(), SearchCriteria.Op.NULL);
         InactiveUnremovedTmpltSearch.done();
 
+        UserDataSearch = createSearchBuilder();
```

Comment:
this might not work well

when we remove a template from a zone, it will not set `removed` field in `vm_template` table.
therefore this search result will contain the templates which have been removed from the zone.

Reply:
updated the query with state "Active"